paddle/phi/kernels/gpu/embedding_grad_kernel.cu

```diff
@@ -61,6 +72,82 @@ __global__ void EmbeddingGrad(T* table,
   }
 }
 
+template <typename T, typename IdT>
+__global__ void EmbeddingGradDeterministic(T* table,
+                                           const T* output,
+                                           const IdT* ids,
+                                           const int64_t K,
+                                           const int64_t D) {
+  extern __shared__ char buf[];
+  T* smem = reinterpret_cast<T*>(buf);
+  T* my_s = smem + WARP_SIZE * threadIdx.y;
+  int* indices_batch =
+      reinterpret_cast<int*>(buf + sizeof(T) * WARP_SIZE * blockDim.y);
+
+  const int s = static_cast<int>(D);
+
+  const int f = threadIdx.x + blockIdx.x * blockDim.x;
```

Comment:
As per torch's code, ```s``` means stride and ```f``` means feature_dim. Could you please rename them or add comments here?

Reply:
Done